### PR TITLE
Ac use zookeeper to preserve its state

### PR DESCRIPTION
**Pull request by obino**
This changes the bootstrap sequence, relying on Zookeeper instead of local disk.

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1286/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1288/

**Comment by obino** [comment]
ocd run https://ocd.appscale.com:8080/job/Daily%20Build/1298/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1300/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1308/

**Comment by obino** [comment]
ocd run https://ocd.appscale.com:8080/job/Daily%20Build/1310/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1314/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1357/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1363/

**Comment by obino** [comment]
ocd run https://ocd.appscale.com:8080/job/Daily%20Build/1394/

**Comment by obino** [comment]
ocd run https://ocd.appscale.com:8080/job/Daily%20Build/1418/

File: AppController/djinnServer.rb [hunk 1]
```diff
@@ -129,20 +127,30 @@ def run_server(server)
...
-  secret = HelperFunctions.get_secret(appscale_dir + "secret.key")
+  begin
+    secret = HelperFunctions.get_secret(appscale_dir + "secret.key")
+  rescue Errno::ENOENT
+    # Ignore the File doesn't exists exception.
+  end
+
   break unless secret.nil?
-  Djinn.log_debug("Still waiting for that secret key...")
+  puts("Still waiting for that secret key...")
```
Comment (cdonati) [inline AppController/djinnServer.rb]: Why was it necessary to change to `puts`?

File: AppController/djinnServer.rb [hunk 2]
```diff
@@ -165,11 +173,26 @@ def run_server(server)
...
+
+# SIGUSR2 is used to signal the orderly shut down of the deployment.
+Signal.trap('SIGUSR2') {
+  puts("Received URS2 signal: stopping the deployment.")
+  response = server.djinn.kill(true, secret)
+  puts("Response to the kill command: #{response}")
+  if !response.start_with?('ERROR')
+    server.shutdown
+  end
+  exit 0
```
Comment (cdonati) [inline AppController/djinnServer.rb]: Could this be a separate PR?

**Comment by obino** [comment]
This code ended up in a different PR and it's already merged with master.
